# Import library for loading, preprocessing data

In [1]:
import pandas as pd
import numpy as np

In [2]:
# load data
data_dir = 'C:/Users/joyh1/Desktop/GitRepo/data/'
filename = 'heart.csv'
data = pd.read_csv(data_dir+filename)
data.shape

(303, 14)

In [3]:
data.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

## Attirbute(Feature) information
- Feature('x' value)
    1. age
    2. sex
    3. chest pain type (4 values)
    4. resting blood pressure
    5. serum cholestoral in mg/dl
    6. fasting blood sugar > 120 mg/dl
    7. resting electrocardiographic results (values 0,1,2)
    8. maximum heart rate achieved
    9. exercise induced angina
    10. oldpeak = ST depression induced by exercise relative to rest
    11. the slope of the peak exercise ST segment
    12. number of major vessels (0-3) colored by flourosopy
    13. thal: 3 = normal; 6 = fixed defect; 7 = reversable defect
---
    
- Label(Class) ('y' value)

    * target


# Data EDA

In [4]:
# 결측치 보기
data.isnull().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

In [5]:
# dtypes 보기
data.dtypes

age           int64
sex           int64
cp            int64
trestbps      int64
chol          int64
fbs           int64
restecg       int64
thalach       int64
exang         int64
oldpeak     float64
slope         int64
ca            int64
thal          int64
target        int64
dtype: object

In [ ]:
# 